### PR TITLE
telemetry.distro.name and telemetry.distro.version

### PR DESCRIPTION
**Pull request by zeitlinger**
- rename `telemetry.auto.name` resource attribute `telemetry.distro.name` 
- add `telemetry.distro.version` resource attribute

### CODE REVIEW

**Review by jsuereth — APPROVED** [review]
Approved, assuming this is already used / prototyped by the Java instrumentation distributions.

**Review by pellared — APPROVED** [review]

**Review by trask — APPROVED** [review]

**Review by breedx-splk — APPROVED** [review]

**Review by mateuszrzeszutek — APPROVED** [review]

**Review by pjanotti — APPROVED** [review]

**Review by rajkumar-rangaraj — APPROVED** [review]

**Comment by Kielek** [comment]
Do you have any recommendation how to call _official_ distributions?

**Review by Kielek — APPROVED** [review]

**Comment by breedx-splk** [comment]
I would be in favor of `telemetry.distro.name` (and version) instead of `auto` because we also have the need in Android, where it's mostly not "auto-instrumentation". We can use this regardless, it's just less fitting with `auto` in the name. I think `distro` is more flexible and applies cleanly to other situations.

**Comment by pellared** [comment]
> I would be in favor of `telemetry.distro.name` (and version) instead of `auto` because we also have the need in Android, where it's mostly not "auto-instrumentation". We can use this regardless, it's just less fitting with `auto` in the name. I think `distro` is more flexible and applies cleanly to other situations.

Then probably we should also rename `telemetry.autp.version` to `telemetry.distro.version`. While I think it is a good idea as the name can even include "auto" to make it clear that the distribution is an automatic instrumentation I would rather prefer to have it as a separate PR.

**Review by AlexanderWert — APPROVED** [review]
I like that proposal.

With a strong +1 on https://github.com/open-telemetry/semantic-conventions/pull/178#issuecomment-1655128785.
So, let's add a similar note for `telemetry.auto.name`.

**Comment by AlexanderWert** [comment]
@zeitlinger How do we want to proceed with that?

Do you mind updating the PR to the proposal from above? (updating to `telemetry.distro.name` + `telemetry.distro.version`): https://github.com/open-telemetry/semantic-conventions/pull/178#issuecomment-1662576602

**Comment by zeitlinger** [comment]
> @zeitlinger How do we want to proceed with that?
> 
> Do you mind updating the PR to the proposal from above? (updating to `telemetry.distro.name` + `telemetry.distro.version`): [#178 (comment)](https://github.com/open-telemetry/semantic-conventions/pull/178#issuecomment-1662576602)

yes, that's a good idea :smile:

**Review by trask — APPROVED** [review]
re-approving (new name `distro`)

@zeitlinger can you update the PR title and description to reflect this change?

**Review by pyohannes — APPROVED** [review]

File: CHANGELOG.md
Comment (pyohannes) [inline CHANGELOG.md:113]: Isn't it the other way around? `telemetry.auto.version` already exists, but not `telemetry.auto.name`.

**Review by pyohannes — CHANGES_REQUESTED** [review]
Looks good. @zeitlinger Please add a new schema file with a schema transformation for this change (like [here](https://github.com/open-telemetry/semantic-conventions/blob/main/schemas/1.21.0#L76)).

**Comment by Kielek** [comment]
> Do you have any recommendation how to call _official_ distributions?
> 
> There is a strict recommendation what to set for sdk.name attribution
> 
> ```
> **[1]:** The OpenTelemetry SDK MUST set the `telemetry.sdk.name` attribute to `opentelemetry`.
> If another SDK, like a fork or a vendor-provided implementation, is used, this SDK MUST set the
> `telemetry.sdk.name` attribute to the fully-qualified class or module name of this SDK's main entry point
> or another suitable identifier depending on the language.
> ```

I would like to rise this topic once more time before merging.

File: CHANGELOG.md
Comment (zeitlinger) [reply CHANGELOG.md:113]: good catch :smile:

**Comment by jsuereth** [comment]
@zeitlinger This still needs a telemetry schema file

File: schemas/1.22.0
Comment (pyohannes) [inline schemas/1.22.0:2]:
```suggestion
schema_url: https://opentelemetry.io/schemas/1.22.0
```

**Review by pyohannes — APPROVED** [review]

**Comment by pyohannes** [comment]
@open-telemetry/specs-semconv-maintainers I guess this can be merged?

**Review by joaopgrassi — CHANGES_REQUESTED** [review]
The schema changes should be reflected in https://github.com/open-telemetry/semantic-conventions/blob/main/schema-next.yaml and not in a new file IIRC. 

Also, was the comment from @Kielek addressed? 

> > Do you have any recommendation how to call _official_ distributions?
> > There is a strict recommendation what to set for sdk.name attribution
> > ```
> > **[1]:** The OpenTelemetry SDK MUST set the `telemetry.sdk.name` attribute to `opentelemetry`.
> > If another SDK, like a fork or a vendor-provided implementation, is used, this SDK MUST set the
> > `telemetry.sdk.name` attribute to the fully-qualified class or module name of this SDK's main entry point
> > or another suitable identifier depending on the language.
> > ```
> 
> I would like to rise this topic once more time before merging.

**Comment by zeitlinger** [comment]
@joaopgrassi @Kielek thanks for the feedback :smile: 

Added a section on naming the official distro & fixed the schema file.

**Review by joaopgrassi — APPROVED** [review]
Giving the broad number of approvals and the latest change (even if minor) I think others need to look at it again.

**Review by Kielek — APPROVED** [review]

File: docs/resource/README.md
Comment (Kielek) [inline docs/resource/README.md:179]: Language infix (like `java`) is kind of duplication of  telemetry.sdk.language but it is fine IMO.

**Review by jsuereth — APPROVED** [review]

File: docs/resource/README.md
Comment (AlexanderWert) [reply docs/resource/README.md:179]: Since it's the distro name (not the distro vendor or so) I think it's fine for the name to be whatever the name of the corresponding GitHub repo (or release artifacts) is. So, IMHO it's fine to have the language in the name.

**Review by AlexanderWert — APPROVED** [review]

File: docs/resource/README.md
Comment (joaopgrassi) [reply docs/resource/README.md:179]: @Kielek all good on your side to close this conversation?

Comment (Kielek) [reply docs/resource/README.md:179]: Sure!

**Pull request merged**
